scipy/linalg/_decomp.py

```diff
@@ -267,7 +267,8 @@ def eig(a, b=None, left=False, right=True, overwrite_a=False,
     return w, vr
 
 
-def eigh(a, b=None, lower=True, eigvals_only=False, overwrite_a=False,
+@_deprecate_positional_args(version="1.14.0")
+def eigh(a, b=None, *, lower=True, eigvals_only=False, overwrite_a=False,
```

Comment:
Are we sure we want the cutoff between positional and keyword arguments _after_ `b`?

Reply:
I would prefer to leave that call to @ilayn. My thought process was that a and b are matrices whereas everything else are options to do with the solve

Reply:
Plus in the docs for other eigenvalue functions a and b are always given as positional and then everything else keyword

Reply:
I'll comment here too, since I see this now.

> Are we sure we want the cutoff between positional and keyword arguments _after_ `b`?

`b` is not a name that carries any information at all, so forcing users to write `b=` is not helpful. The purpose of keyword-only parameters is to enforce better practices and help users write clearer code. Compare:

- `eigh(a, b)` vs `eigh(a, b=b)` - not a gain (if anything, I'd prefer the former)
- `eigh(a, b, False)` vs. `eigh(a, b, lower=False)` - now we're actually getting a gain in code clarity, because `lower` can be understood

Reply:
Indeed, what Ralf said and +1 to cut after `b`.